src/libs/OptionsListUtils.ts

```diff
@@ -1997,6 +2011,73 @@ function formatSectionsFromSearchTerm(
     };
 }
 
+function filterOptions(options: GetOptions, searchValue: string): ReportUtils.OptionData[] {
+    const searchTerms = searchValue.split(' ');
+
+    const reportsByType = options.recentReports.reduce<ReportTypesOptionData>(
+        (acc, option) => {
+            if (!!option.isChatRoom || !!option.isPolicyExpenseChat) {
+                acc.chatRoomsAndPolicyExpenseChats.push(option);
+            } else {
+                acc.reports.push(option);
+            }
+
+            return acc;
+        },
+        {
+            chatRoomsAndPolicyExpenseChats: [],
+            reports: [],
+            personalDetails: options.personalDetails,
+        },
+    );
+    const createFilter = (items: ReportUtils.OptionData[], keys: ReadonlyArray<KeyOption<ReportUtils.OptionData>>, term: string) =>
+        filterArrayByMatch(items, term, {
+            keys,
+            strict: true,
+        });
+
+    // The regex below is used to remove dots only from the local part of the user email (local-part@domain)
+    // so that we can match emails that have dots without explicitly writing the dots (e.g: fistlast@domain will match first.last@domain)
+    const emailRegex = /\.(?=[^\s@]*@)/g;
```

Comment:
probably you can initialize it outside of the function

Reply:
hmm, I think it might negatively affect the performance, because we would have to call it for each value under a specified keys. With current approach, we can limit those checks only to the keys we need.